From tytso@mit.edu  Wed Feb 18 11:11:31 2009
From: Mark Fasheh <mfasheh@suse.com>
Date: Tue, 17 Feb 2009 10:58:35 -0500
Subject: jbd2: Add BH_JBDPrivateStart
To: stable@kernel.org
Cc: Mark Fasheh <mfasheh@suse.com>, linux-ext4@vger.kernel.org
Message-ID: <1234886324-15105-16-git-send-email-tytso@mit.edu>


From: Mark Fasheh <mfasheh@suse.com>

(cherry picked from commit e97fcd95a4778a8caf1980c6c72fdf68185a0838)

Add this so that file systems using JBD2 can safely allocate unused b_state
bits.

In this case, we add it so that Ocfs2 can define a single bit for tracking
the validation state of a buffer.

Acked-by: "Theodore Ts'o" <tytso@mit.edu>
Signed-off-by: Mark Fasheh <mfasheh@suse.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 include/linux/jbd2.h |    1 +
 1 file changed, 1 insertion(+)

--- a/include/linux/jbd2.h
+++ b/include/linux/jbd2.h
@@ -329,6 +329,7 @@ enum jbd_state_bits {
 	BH_State,		/* Pins most journal_head state */
 	BH_JournalHead,		/* Pins bh->b_private and jh->b_bh */
 	BH_Unshadow,		/* Dummy bit, for BJ_Shadow wakeup filtering */
+	BH_JBDPrivateStart,	/* First bit available for private use by FS */
 };
 
 BUFFER_FNS(JBD, jbd)
